src/test/java/com/commercetools/sync/cartdiscounts/utils/CartDiscountUpdateActionUtilsTest.java

```diff
@@ -0,0 +1,1473 @@
+
+    @Test
+    void buildChangeSortOrderUpdateAction_WithDifferentValues_ShouldBuildUpdateAction() {
+        final CartDiscount oldCartDiscount = mock(CartDiscount.class);
+        when(oldCartDiscount.getSortOrder()).thenReturn("0.1");
+
+        final CartDiscountDraft newCartDiscountDraft = mock(CartDiscountDraft.class);
+        final String sortOrder = "0.3";
+        when(newCartDiscountDraft.getSortOrder()).thenReturn(sortOrder);
+
+        final Optional<UpdateAction<CartDiscount>> changeChangeSortOrderUpdateAction =
+                buildChangeSortOrderUpdateAction(oldCartDiscount, newCartDiscountDraft);
+
+        assertThat(changeChangeSortOrderUpdateAction).contains(ChangeSortOrder.of(sortOrder));
+    }
+
+    @Test
+    void buildChangeSortOrderUpdateAction_WithSameValues_ShouldNotBuildUpdateAction() {
+        final String sortOrder = "0.1";
+
+        final CartDiscount oldCartDiscount = mock(CartDiscount.class);
+        when(oldCartDiscount.getSortOrder()).thenReturn(sortOrder);
+
+        final CartDiscountDraft newCartDiscountDraft = mock(CartDiscountDraft.class);
+        when(newCartDiscountDraft.getSortOrder()).thenReturn(sortOrder);
+
+        final Optional<UpdateAction<CartDiscount>> changeChangeSortOrderUpdateAction =
+                buildChangeSortOrderUpdateAction(oldCartDiscount, newCartDiscountDraft);
+
+        assertThat(changeChangeSortOrderUpdateAction).isNotPresent();
+    }
+
+    @Test
+    void buildChangeSortOrderUpdateAction_WithJustNewSortOrderHasNullValue_ShouldBuildUpdateAction() {
+        final CartDiscount oldCartDiscount = mock(CartDiscount.class);
+        when(oldCartDiscount.getSortOrder()).thenReturn("0.1");
+
+        final CartDiscountDraft newCartDiscountDraft = mock(CartDiscountDraft.class);
+        when(newCartDiscountDraft.getSortOrder()).thenReturn(null);
+
+        final Optional<UpdateAction<CartDiscount>> changeChangeSortOrderUpdateAction =
+                buildChangeSortOrderUpdateAction(oldCartDiscount, newCartDiscountDraft);
+
+        assertThat(changeChangeSortOrderUpdateAction).contains(ChangeSortOrder.of(null));
+    }
+
+    @Test
+    void buildChangeSortOrderUpdateAction_WithJustOldSortOrderHasNullValue_ShouldBuildUpdateAction() {
+        final CartDiscount oldCartDiscount = mock(CartDiscount.class);
+        when(oldCartDiscount.getSortOrder()).thenReturn(null);
+
+        final CartDiscountDraft newCartDiscountDraft = mock(CartDiscountDraft.class);
+        final String newSortOrder = "0.3";
+        when(newCartDiscountDraft.getSortOrder()).thenReturn(newSortOrder);
+
+        final Optional<UpdateAction<CartDiscount>> changeChangeSortOrderUpdateAction =
+                buildChangeSortOrderUpdateAction(oldCartDiscount, newCartDiscountDraft);
+
+        assertThat(changeChangeSortOrderUpdateAction).contains(ChangeSortOrder.of(newSortOrder));
+    }
+
+    @Test
+    void buildChangeSortOrderUpdateAction_WithBothSortOrderHaveNullValues_ShouldNotBuildUpdateAction() {
```

Comment:
but it seems that it is also not optional from docs: https://docs.commercetools.com/http-api-projects-cartDiscounts#cartdiscountdraft